### PR TITLE
Bring latest changes from SZ8

### PR DESCRIPTION
**Pull request by ktonga**

### CODE REVIEW

**Comment by ktonga** [comment]
Hey @jdegoes ZIO is up2date! Next step, rename root package and publish snapshots. I'll start the changes on `scalaz/scalaz` after that.

File: core/shared/src/main/scala/scalaz/effect/IO.scala [hunk 1]
```diff
@@ -376,29 +376,48 @@ sealed abstract class IO[E, A] { self =>
...
    * necessarily at the specified interval.
    */
-  final def repeatFixed[B](interval: Duration): IO[E, B] =
+  final def repeatFixed(interval: Duration): IO[E, A] =
     repeatFixed0(IO.sync(System.nanoTime()))(interval)
 
-  final def repeatFixed0[B](nanoTime: IO[Nothing, Long])(interval: Duration): IO[E, B] =
-    IO.flatten(nanoTime[E].flatMap { start =>
-      val gapNs = interval.toNanos
+  final def repeatFixed0(nanoTime: IO[Nothing, Long])(interval: Duration): IO[E, A] = {
```
Comment (ktonga) [inline core/shared/src/main/scala/scalaz/effect/IO.scala]: @jdegoes most of the changes are pure git-fu, either whole new files or trivial changes. This is the only thing i manually changed and can be lost within the big chunk of diffs.

File: benchmarks/src/main/scala/scalaz/effect/BubbleSortBenchmarks.scala [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import IOBenchmarks.unsafePerformIO
+
+@State(Scope.Thread)
+@BenchmarkMode(Array(Mode.Throughput))
+@OutputTimeUnit(TimeUnit.SECONDS)
+class BubbleSortBenchmarks {
+  @Param(Array("1000"))
+  var size: Int = _
+
+  def createTestArray: Array[Int] = Range.inclusive(1, size).toArray.reverse
```
Comment (jdegoes) [inline benchmarks/src/main/scala/scalaz/effect/BubbleSortBenchmarks.scala]: Why aren't all of these importing `Array`, `Int`, etc? We turned off no implicit prelude in the main scalaz.

File: core/shared/src/main/scala/scalaz/effect/package.scala [hunk 3]
```diff
@@ -2,7 +2,15 @@
...
 
+import scala.AnyRef
+
 package object effect {
+  type Error            = java.lang.Error
+  type Exception        = java.lang.Exception
+  type InternalError    = java.lang.InternalError
+  type OutOfMemoryError = java.lang.OutOfMemoryError
+  type RuntimeException = java.lang.RuntimeException
+  type Throwable        = java.lang.Throwable
```
Comment (jdegoes) [inline core/shared/src/main/scala/scalaz/effect/package.scala]: I'm not sure we want to expose these... ideally not.

File: benchmarks/src/main/scala/scalaz/effect/BubbleSortBenchmarks.scala (hunk 2 above)
Comment (ktonga) [reply benchmarks/src/main/scala/scalaz/effect/BubbleSortBenchmarks.scala]: I didn't import that change, to make it easier to use this lib in a non-scalaz environment. We are using std everywhere as opposed to scalaz core where using our own data types is preferred.

File: core/shared/src/main/scala/scalaz/effect/package.scala (hunk 3 above)
Comment (ktonga) [reply core/shared/src/main/scala/scalaz/effect/package.scala]: I just ported that 1:1 from sz8. You want me to remove it?

Comment (ktonga) [reply core/shared/src/main/scala/scalaz/effect/package.scala]: I removed it. It was introduced due to the no-imports thing but we don't need it anyway.

File: core/shared/src/main/scala/scalaz/effect/IO.scala (hunk 1 above)
Comment (ktonga) [reply core/shared/src/main/scala/scalaz/effect/IO.scala]: I've restored the type param

File: benchmarks/src/main/scala/scalaz/effect/BubbleSortBenchmarks.scala (hunk 2 above)
Comment (jdegoes) [reply benchmarks/src/main/scala/scalaz/effect/BubbleSortBenchmarks.scala]: I think we should make compiler options as consistent as possible. Obviously we can't enforce Scalazzi here and we can't depend on any other project, but strict compiler options consistent with Scalaz will help ensure uniform coding standards across the Scalaz projects.

Comment (ktonga) [reply benchmarks/src/main/scala/scalaz/effect/BubbleSortBenchmarks.scala]: Ok, I'll do it in a separate PR, it's not a priority to get snapshots published right now IMHO.

Comment (jdegoes) [reply benchmarks/src/main/scala/scalaz/effect/BubbleSortBenchmarks.scala]: Agreed.

**Comment by jdegoes** [comment]
@ktonga You all done with changes & ready for a merge?

**Comment by ktonga** [comment]
I think so

**Comment by jdegoes** [comment]
@wi101 If you see anything, just create a ticket for it. I will merge this now.

**Comment by wi101** [comment]
@jdegoes sorry for the late reply, Ok I will see now

**Pull request merged**